### DISCUSSION TITLE
Any tips for a beginner creating an own model in new language using Coqui?

### DISCUSSION THREAD

**Discussion by coodere** [coqui-ai/TTS#2367] [General Q&A]
I'm trying to train my own model and language in Filipino for my research. I've followed the LJSpeech dataset format and was already training my own dataset (a total of 477 samples) through collab.research.google.com using [this notebook](https://github.com/coqui-ai/TTS/blob/dev/notebooks/Tutorial_2_train_your_first_TTS_model.ipynb) through VITS_TTS recipe. For reference here is my recent configuration
```
audio_config = VitsAudioConfig(
sample_rate=22050, win_length=1024, hop_length=256, num_mels=80, mel_fmin=0, mel_fmax=None
)
characters_config = CharactersConfig(
    pad = '<PAD>',
    eos = '।', #'<EOS>', #'।',
    bos = '<BOS>',# None,
    blank = '<BLNK>',
    phonemes = None,
    characters =  "ABCDEFGHIJKLMNÑOPQRSTUVWXYZabcdefghijklmnñopqrstuvwxyz",
    punctuations = "!'(),-.:;? ",
)
config = VitsConfig(
    audio=audio_config,
    run_name="vits2_ljspeech",
    batch_size=32,
    eval_batch_size=16,
    batch_group_size=5,
    num_loader_workers=8,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=500,
    text_cleaner="collapse_whitespace",
    use_phonemes=False,
#    phoneme_language="en-us",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    compute_input_seq_cache=True,
    print_step=25,
    print_eval=True,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
    cudnn_benchmark=False,
    characters = characters_config,
)
```
Here is a simple output that the model is giving.

https://user-images.githubusercontent.com/39657228/221407042-0c2b12bc-cf65-4fb4-bc76-c9498c019ee9.mp4

Any thoughts/recommendations for this model to better improve would be appreciated very much.
I was also wondering if anyone can walk me through creating/detecting my own phoneme for this language, as Coqui doesn't support my language just yet. Thanks!

**Comment by JohnGarrison78** [comment]
I too would like a better explanation of how phonemes and phonemizers are used in the system. I'd like to create a TTS model for a  series of conlangs I'm creating. I can easily convert the text output to IPA, but I'm unsure if that's all the phonemizer needs to do. There's a blog post celebrating the bringing of the various African languages to the system, but it doesn't go into a lot of detail of how it was accomplished. The same is true, to a lesser extant, of tokenizers. I've seen a few documents and tutorials, but they all just give a list of instructions rather than explaining why the steps are taken and what's happening underneath the hood, which makes it hard to apply to other use cases.

**Comment by jangotek** [comment]
any luck bro for Filipino language? looking forward also...
